### ISSUE TITLE
registeredComponentsOnly is not included in vue/component-name-in-template-casing rule

### ISSUE THREAD

**Issue by kpping** [vuejs/eslint-plugin-vue#762]
HI, I have tried to use `vue/component-name-in-template-casing` rule with `registeredComponentsOnly` option, [ref](https://eslint.vuejs.org/rules/component-name-in-template-casing.html), but it seems like `registeredComponentsOnly` haven't included in 5.1.0 yet.

**Tell us about your environment**	
* **ESLint version: 5.12.0**	
* **eslint-plugin-vue version: 5.1.0**	
* **Node version: 10.15.0**	

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
  },
  parserOptions: {
    parser: 'babel-eslint',
  },
  extends: [
    'airbnb-base',
    'plugin:vue/recommended',
    'plugin:prettier/recommended',
    'prettier',
    'prettier/vue',
  ],
  // required to lint *.vue files
  plugins: [
    'prettier',
    'vue',
  ],
  // add your custom rules here
  rules: {
    'import/no-extraneous-dependencies': 'off',
    'import/no-unresolved': 'off',
    'import/prefer-default-export': 'off',
    'max-len': 'off',
    'no-console': 'warn',
    'no-plusplus': ['error', {
      allowForLoopAfterthoughts: true,
    }],
    "vue/component-name-in-template-casing": ["error", "PascalCase", {
      registeredComponentsOnly: true,
      ignores: [],
    }],
  },
}
```

**What did you do?**	

```vue
<template lang="html">
  <v-app>
    <global-error-snackbar/>
    <nuxt />
  </v-app>
</template>

<script>
import GlobalErrorSnackbar from '~/components/GlobalErrorSnackbar.vue';

export default {
  components: {
    GlobalErrorSnackbar,
  },
};
</script>

<style lang="scss"></style>
```

**What did you expect to happen?**
```vue
<template lang="html">
  <v-app>
    <GlobalErrorSnackbar/>
    <nuxt />
  </v-app>
</template>
```

**What actually happened?**
```bash
$ eslint --ext .js,.vue .
Error: some_path_to_my_eslintrc/.eslintrc.js:
	Configuration for rule "vue/component-name-in-template-casing" is invalid:
	Value {"registeredComponentsOnly":true,"ignores":[]} should NOT have additional properties.
```

**Comment by ota-meshi** [comment]
Probably the following issues are also caused by [https://eslint.vuejs.org](https://eslint.vuejs.org) document.

- #764
- https://github.com/vuejs/eslint-plugin-vue/pull/680#issuecomment-453524512

**Comment by kpping** [comment]
`registeredComponentsOnly` has been added, tested in eslint-plugin-vue version: ^5.2.2

thank you.
